### PR TITLE
Breaking: Remove redundant data and simplify reported messages

### LINKED ISSUE #3693: Remove URLs and code snippets from messages
We should review and simplify messages reported from hints to clean up the UX across entry points.

Why?
1. Redundant with URLs and code snippets provided in report metadata
2. Causes some messages to be VERY long
3. Prevents collapsing similar messages

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [x] Added/Updated related tests.

## Short description of the change(s)

The goal of simplifying reported messages is two-fold:
1. Make messages easier to read by eliminating long blocks of markup and URLs (which are repeated in the expanded details of a message).
2. Increase the ability for similar messages to be collapsed, giving users an improved overview of the number of unique issue types reported.

I've also made editorial changes to messages for clarity and consistency as I see opportunities. Feel free to provide additional feedback to this end. Since we're changing the messages anyway, now is a great time to ensure we're happy with the precise wording.

Fixes #3693

### CODE REVIEW

**Comment by molant** [comment]
Haven't reviewed yet, but are there any other breaking changes we want to do while we are at it?

**Comment by antross** [comment]
> Haven't reviewed yet, but are there any other breaking changes we want to do while we are at it?

None that are top-of-mind. Happy to debate to what degree we want to consider this breaking too. By default this will only be breaking at the individual hint level. 

Also, I personally don't feel the exact text of a message is a contract at the semver level (especially once you throw i18n in the mix), but I erred on the conservative side with marking these as breaking for now given our prior conversations. I figure we can debate this a bit before actually merging the changes.

That said, we currently don't expose the concept of an error code, so there's nothing more granular than a hint ID to use to recognize an individual error unless you look at the message. Perhaps that's something we should consider.

**Comment by antross** [comment]
@hxlnt, @johnemau, @molant: This is ready for review when you get the chance.

If you want you can self-host the changes by side-loading this updated version of the browser extension:
* [webhint-2.1.1.zip](https://github.com/webhintio/hint/files/4600799/webhint-2.1.1.zip)

File: packages/hint-http-cache/src/_locales/en/messages.json [hunk 1]
```diff
@@ -5,42 +5,42 @@
     },
     "directiveInvalid": {
         "description": "Report message when the directives are invalid",
-        "message": "The $1 $2 $3 invalid"
+        "message": "A 'cache-control' header contains invalid directives: $1"
```
Comment (molant) [inline packages/hint-http-cache/src/_locales/en/messages.json:8]: Should we wrap the directives with `'`  or `"` for consistency and make it even clearer?

File: packages/hint-no-http-redirects/src/_locales/en/messages.json [hunk 2]
```diff
@@ -7,12 +7,8 @@
...
         "message": "Avoid HTTP redirects"
     },
-    "redirectDectected": {
-        "description": "Report message when a redirect are detected",
-        "message": "$1 redirect detected for '$2' (max is $3)."
-    },
     "redirectsDectected": {
         "description": "Report message when a redirect are detected",
-        "message": "$1 redirects detected for '$2' (max is $3)."
+        "message": "Too many HTTP redirects detected (max is $1)."
```
Comment (molant) [inline packages/hint-no-http-redirects/src/_locales/en/messages.json:12]: 👏

File: packages/hint-http-cache/src/_locales/en/messages.json (hunk 1 above)
Comment (antross) [reply packages/hint-http-cache/src/_locales/en/messages.json:8]: Were you thinking of quoting the directives as a group or individually (e.g. when there's more than one)?

Comment (molant) [reply packages/hint-http-cache/src/_locales/en/messages.json:8]: I think group should be fine as they are part of the same header value (and it should be easier to implement)? Don't have a strong opinion though.

Comment (antross) [reply packages/hint-http-cache/src/_locales/en/messages.json:8]: Ok, I went with quoting the individual values.

**Comment by antross** [comment]
@hxlnt do you have any feedback or does this look good to merge?

@molant do you have a preference whether this is squashed into a single breaking commit (since it's effectively the same intent across packages) or remains a series of per-package commits?

**Comment by molant** [comment]
Squashing is fine 👍

**Pull request merged**